onchain/src/social_account.cairo

```diff
@@ -0,0 +1,73 @@
+use starknet::{ContractAddress, get_caller_address, get_contract_address, contract_address_const};
+
+
+#[starknet::interface]
+pub trait ISocialPayAccount<TContractState> {
+    fn get_public_key(self: @TContractState) -> u256;
+}
+
+
+#[starknet::contract]
+pub mod SocialPayAccount {
+    #[storage]
+    struct Storage {
+        #[key]
+        public_key: u256
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        AccountCreated: AccountCreated,
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct AccountCreated {
+        #[key]
+        public_key: u256
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, public_key: u256) {
+        self.public_key.write(public_key);
+    }
+
+    #[abi(embed_v0)]
+    impl SocialPayAccount of super::ISocialPayAccount<ContractState> {
+        fn get_public_key(self: @ContractState) -> u256 {
+            self.public_key.read()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use starknet::{
+        ContractAddress, get_caller_address, get_contract_address, contract_address_const
+    };
+    use snforge_std as snf;
+    use snforge_std::{declare, ContractClassTrait};
+    use super::{ISocialPayAccountDispatcher, ISocialPayAccountDispatcherTrait};
+
+    const public_key: u256 = 45;
+
+    fn deploy_social_account() -> ContractAddress {
+        let contract = declare("SocialPayAccount");
+
```

Comment:
`AccountCreated` event emission should be verified here.

Reply:
you mean, check that AccountCreated was actually emmited

Reply:
how do check that event is actually emitted inside a function @maciejka

Reply:
With starknet foundry: https://foundry-rs.github.io/starknet-foundry/testing/testing-events.html

Reply:
i have this 

    fn deploy_social_account() -> ContractAddress {
        let contract = declare("SocialPayAccount");

        let mut social_account_calldata = array![];
        public_key.serialize(ref social_account_calldata);

        let address = contract.precalculate_address(@social_account_calldata);
        
        let deployed_contract_address =  contract.deploy(@social_account_calldata).unwrap();
        
        let mut spy = spy_events(SpyOn::One(address));
        spy.assert_emitted(@array![
        (
            contract_address,
            AccountCreated { public_key }
            
        )
    ]);

       //  contract.deploy(@social_account_calldata).unwrap()
       deployed_contract_address
    }